jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/MethodScope.java

```diff
@@ -171,11 +177,49 @@ public boolean isGetter() {
     public <A extends Annotation> A getAnnotation(Class<A> annotationClass) {
         A annotation = super.getAnnotation(annotationClass);
         if (annotation == null) {
-            annotation = this.getRawMember().getAnnotatedReturnType().getAnnotation(annotationClass);
+            annotation = findGetterCandidates(this.getRawMember()).stream().map(m -> {
+                return Optional.ofNullable(m.getAnnotation(annotationClass))
+                        .orElse(m.getAnnotatedReturnType().getAnnotation(annotationClass));
```

Comment:
**suggestion**: Two challenges I see here:
> 1. False assumption: When `MethodScope.getAnnotation()` is invoked, all targeted annotations may be inherited.

I'm afraid that doesn't hold true for all types of annotations, i.e., we'd probably have to put this extra logic into the `TypeContext` and allow this to be decided by the caller (on a per-annotation basis) by offering a new/alternative method for it. And then adjust the standard modules where appropriate (e.g. for javax-validation and its jakarta equivalent).

> The "fake container item scope" is not being considered.

E.g., in case of a method returning `List<@NotEmpty String>`, the library is building a schema for the same `MethodScope` twice: first iteration for the `List` and the second iteration for the `String` container item type.  The latter should also be covered in case of overridden methods I guess.